client/tests/integration/queries/asset_definition.rs

```diff
@@ -0,0 +1,41 @@
+#![allow(clippy::restriction)]
+
+use std::str::FromStr as _;
+
+use eyre::Result;
+use iroha_client::client;
+use iroha_data_model::prelude::*;
+use test_network::Peer as TestPeer;
+
+#[test]
+fn find_asset_definition_by_id() -> Result<()> {
```

Comment:
Since this is not what I asked you to do, I would like to know why you decided not to add the query to an existing test.

Reply:
OK, now I got you. So I'm going to move this test into the closest module ([account.rs](https://github.com/hyperledger/iroha/blob/iroha2-dev/client/tests/integration/queries/account.rs)) like this:
```rs
    // Registering accounts
    let register_accounts = accounts
        .iter()
        .skip(1) // Alice has already been registered in genesis
        .cloned()
        .map(|account_id| RegisterBox::new(Account::new(account_id, [])).into())
        .collect::<Vec<_>>();
    test_client.submit_all_blocking(register_accounts)?;

   // HERE
===>  let received_asset_definition = test_client.request(client::asset::definition_by_id(definition_id))?;
===>  assert_eq!(received_asset_definition, asset_definition.build());

    let mint_asset = accounts
        .iter()
        .cloned()
        .map(|account_id| <Asset as Identifiable>::Id::new(definition_id.clone(), account_id))
        .map(|asset_id| MintBox::new(1_u32, asset_id).into())
        .collect::<Vec<_>>();
    test_client.submit_all_blocking(mint_asset)?;

   // HERE AFTER MINTING
===>   let received_asset_definition = test_client.request(client::asset::definition_by_id(definition_id))?;
===>   assert_eq!(received_asset_definition, asset_definition.build());

    let accounts = HashSet::from(accounts);
```

Reply:
I now realize that this PR only needs to replace `FindAllAssetsDefinitions` with `FindAssetDefinitionById` in [this Torii unit test](https://github.com/hyperledger/iroha/blob/1f1c8d79524b4af3536f3505d103014d86304ec9/cli/src/torii/tests.rs#L414-L438)

Reply:
Yeah seems this test needs to replace but its only on the server-side. I also added a new client-side method so I think it also should be tested.